core/modules/replica/IngestServerConnection.cc

```diff
@@ -0,0 +1,520 @@
+            LOGS(_log, LOG_LVL_ERROR, context << __func__ << "  " << error);
+            _failed(error);
+        }
+    } else {
+
+        size_t maxRows = 1;
+        if (rowSize != 0) maxRows = max(maxRows, networkBufSizeBytes / rowSize);
+        _sendReadyToReadData(maxRows);
+    }
+}
+
+
+void IngestServerConnection::_reply(ProtocolIngestResponse::Status status,
+                                    string const& msg,
+                                    size_t maxRows) {
+    ProtocolIngestResponse response;
+    response.set_status(status);
+    response.set_error(msg);
+    response.set_max_rows(maxRows);
+
+    _bufferPtr->resize();
+    _bufferPtr->serialize(response);
+
+    _sendResponse();
+}
+
+
+void IngestServerConnection::_loadDataIntoTable() {
+
+    LOGS(_log, LOG_LVL_DEBUG, context << __func__);
+
+    // ATTENTION: the data loading metod used in this implementation requires
+    // that the MySQL server has (at least) the read-only access to files in
+    // a folder in which the CSV file will be stored by this server. So, make
+    // proper adjustments to a configuration of the Replication system.
+
+    database::mysql::Connection::Ptr conn;
+    try {
+        
+        conn = database::mysql::Connection::open(database::mysql::ConnectionParams(
+            _workerInfo.dbHost,
+            _workerInfo.dbPort,
+            _workerInfo.dbUser,
+            _serviceProvider->config()->qservWorkerDatabasePassword(),
+            ""
+        ));
+
+        string const sqlDatabase = conn->sqlId(_databaseInfo.name);
+        string const sqlPartition = conn->sqlId("p" + to_string(_transactionId));
+
+        vector<string> statements;
+
+        if (_isPartitioned) {
+            
+            // Chunked tables are created from the prototype table which is expected
+            // to exist in the database before attempting data loading.
+
+            string const sqlProtoTable = sqlDatabase + "." + conn->sqlId(_table);
+            string const sqlTable = sqlDatabase + "." + conn->sqlId(_table + "_" + to_string(_chunk));
+            string const sqlFullOverlapTable  =
+                sqlDatabase + "." + conn->sqlId(_table + "FullOverlap_" + to_string(_chunk));
+
+            string const sqlDummyChunkTable =
+                sqlDatabase + "." + conn->sqlId(_table + "_" + to_string(lsst::qserv::DUMMY_CHUNK));
+
+            string const sqlOverlapDummyChunkTable =
+                sqlDatabase + "." + conn->sqlId(_table + "FullOverlap_" + to_string(lsst::qserv::DUMMY_CHUNK));
+
+            vector<string> const tablesToBeCreated = {
+                sqlTable,
+                sqlFullOverlapTable,
+                sqlDummyChunkTable,
+                sqlOverlapDummyChunkTable
+            };
+            for (auto&& table: tablesToBeCreated) {
+                statements.push_back(
+                    "CREATE TABLE IF NOT EXISTS " + table + " LIKE " + sqlProtoTable
```

Comment:
Yes, it does.

Reply:
My first reaction is that it should not.

Reply:
Why?

Reply:
All other (chunked) tables at workers have this column. The column could (after the ingest is done) be eliminated if needed. Though, it's better to keep them for the bookkeeping and Q&A purposes to investigate problems with ingested data later.

Reply:
This column is no different from `chunkId` and `subChunkId` which are present in both prototype and chunked tables.

Reply:
If someone else want to use that schema for any other purpose they'd have to know that there are columns in that schema which are specific to your implementation of partitioning mechanism (and that mechanism could change too). It would probably be better to have an original schema without any implementation-specific details and extend schema when needed.